nilearn/glm/tests/test_first_level.py

```diff
@@ -272,6 +272,36 @@ def test_high_level_glm_with_data_with_mask(tmp_path):
     )
 
 
+def test_fmri_inputs_type_data_smoke(tmp_path):
+    """Test processing of FMRI inputs with path, str or nifti for data."""
+    shapes = ((9, 9, 9, 6),)
+    mask, func_img, des = write_fake_fmri_data_and_design(
+        shapes, file_path=tmp_path
+    )
+    FirstLevelModel(mask_img=mask).fit(func_img[0], design_matrices=des[0])
+    FirstLevelModel(mask_img=mask).fit(
+        [Path(func_img[0])], design_matrices=des[0]
+    )
+    FirstLevelModel(mask_img=mask).fit(
+        load(func_img[0]), design_matrices=des[0]
+    )
+
+
+def test_fmri_inputs_type_design_matrices_smoke(tmp_path):
+    """Test processing of FMRI inputs with path, str for design matrix."""
+    shapes = ((9, 9, 9, 6),)
```

Comment:
In general, I avoid having identical shapes values, because this fails to detect if the code does not handle dimensions well.

Reply:
Ah yeah that makes sense.
Will actually then update some of the pytest fixtures accordingly.